File: frontend/src/types/revenue.ts
/** Matches backend/models/revenue.py */

export interface RevenueTrendPoint {
  period_start: string
  period_label: string | null
  deal_revenue: number | null
  subscription_mrr_booked: number | null
  subscription_arr_booked: number | null
  invoice_amount: number | null
  payments_received: number | null
}

export interface RevenueTrendsResponse {
  period: string
  points: RevenueTrendPoint[]
}

export interface RevenueForecastPoint {
  forecast_date: string
  predicted_revenue: number
  lower_bound: number
  upper_bound: number
  model_version: string
  generated_at: string
}

export interface RevenueHistoricalPoint {
  month: string
  invoice_amount: number
}

export interface RevenueForecastResponse {
  historical_context: RevenueHistoricalPoint[]
  forecast: RevenueForecastPoint[]
}

export interface RevenueByRegionRow {
  region_name: string
  deal_revenue: number | null
  invoice_amount: number | null
  payments_received: number | null
}

export interface RevenueByRegionResponse {
  rows: RevenueByRegionRow[]
}

File: backend/models/revenue.py
from datetime import date, datetime
from typing import List, Optional

from pydantic import BaseModel


class RevenueTrendPoint(BaseModel):
    period_start: date
    period_label: Optional[str] = None
    deal_revenue: Optional[float] = None
    subscription_mrr_booked: Optional[float] = None
    subscription_arr_booked: Optional[float] = None
    invoice_amount: Optional[float] = None
    payments_received: Optional[float] = None


class RevenueTrendsResponse(BaseModel):
    period: str
    points: List[RevenueTrendPoint]


class RevenueForecastPoint(BaseModel):
    forecast_date: date
    predicted_revenue: float
    lower_bound: float
    upper_bound: float
    model_version: str
    generated_at: datetime


class RevenueHistoricalPoint(BaseModel):
    month: date
    invoice_amount: float


class RevenueForecastResponse(BaseModel):
    historical_context: List[RevenueHistoricalPoint]
    forecast: List[RevenueForecastPoint]
